noir-projects/aztec-nr/aztec/src/event/event_interface.nr

```diff
@@ -14,50 +14,136 @@ use dep::protocol_types::{
+/// |                                                                                                        |
+/// | This combination is useful when we either trust the sender to behave or the sender is incentivized to  |
+/// | deliver the message. To give you an example let's consider a payment app. When you are buying          |
+/// | groceries, the merchant can consider the payment successful once he receives the payment message, can  |
+/// | successfully decrypt it and see that its contents correspond to a valid token note.                    |
+/// +--------------------------------------------------------------------------------------------------------+
+///
+///
+/// Other combinations like unconstrained encryption, unconstrained tagging, private log stream don't seem
+/// to be meaningful because they are costlier than combination 2 above while providing no meaningful
+/// additional guarantees.
+pub struct MessageDeliveryEnum {
+    /// Constrained encryption + constrained tagging + private log stream
     ///
-    /// Only use in scenarios where the recipient not receiving the message is an acceptable outcome (e.g. because the
-    /// sender is somehow motivated to ensure the recipient learns of it).
-    pub NO_CONSTRAINTS: u8,
-    /// The contents of the log and its encryption are constrained. The tag (and therefore whether the recipient is
-    /// actually able to find the message) is not.
+    /// This gives us an on-chain guarantee that a recipient is able to discover and receive a message. See
+    /// combination 1 above for details.
+    pub CONSTRAINED_ONCHAIN: u8,
+
+    /// Unconstrained encryption + offchain message
     ///
-    /// Only use in scenarios where the recipient not receiving the message is an acceptable outcome (e.g. because the
-    /// sender is somehow motivated to ensure the recipient learns of it).
-    // TODO(#14565): This variant requires for tagging to also be constrained, as it is otherwise useless.
-    pub CONSTRAINED_ENCRYPTION: u8,
+    /// Useful when we trust the sender or the sender is incentivized to deliver the message. See combination 2
+    /// above for details.
+    pub UNCONSTRAINED_OFFCHAIN: u8,
+
+    /// Unconstrained encryption + unconstrained tagging + private log stream
+    ///
+    /// As mentioned above, this combination is not meaningful but we keep it here as it's currently used in the
+    /// codebase.
+    /// TODO: Consider dropping this.
+    pub UNCONSTRAINED_ONCHAIN: u8,
 }
 
-pub global PrivateLogContent: PrivateLogContentConstraintsEnum = PrivateLogContentConstraintsEnum {
-    NO_CONSTRAINTS: 1,
-    CONSTRAINED_ENCRYPTION: 2,
-    // TODO: add constrained tagging and constrained handshaking
+pub global MessageDelivery: MessageDeliveryEnum = MessageDeliveryEnum {
+    CONSTRAINED_ONCHAIN: 1,
+    UNCONSTRAINED_OFFCHAIN: 2,
+    UNCONSTRAINED_ONCHAIN: 3,
```

Comment:
This ordering is odd, I'd expect for this to be ordered with either increasing or decreasing guarantees, but this is neither. 1 is most guarantees, 2 is least guarantees, 3 is in the middle.